docs/source/protocols.rst

```diff
@@ -319,7 +319,8 @@ member:
    batch_proc([], bad_cb)   # Error! Argument 2 has incompatible type because of
                             # different name and kind in the callback
 
-Callback protocols and :py:data:`~typing.Callable` types can be used interchangeably.
+Callback protocols and :py:data:`~typing.Callable` types can be used interchangeably (altough using
+:py:data:`~typing.Type` on a :py:data:`~typing.Callable` might lead to unexpected results).
```

Comment:
Without going into details, this is the reason I've added this part: https://mypy-play.net/?mypy=1.2.0&python=3.11&gist=25c2697cb3c308ae1609b0f6dc9ff0ff